src/main/java/net/minecraftforge/server/command/CommandSetDimension.java

```diff
@@ -0,0 +1,130 @@
+    }
+
+    @Override
+    public List<String> getAliases()
+    {
+        return Collections.singletonList("setdim");
+    }
+
+    @Override
+    public String getUsage(ICommandSender sender)
+    {
+        return "commands.forge.setdim.usage";
+    }
+
+    @Override
+    public List<String> getTabCompletions(MinecraftServer server, ICommandSender sender, String[] args, @Nullable BlockPos targetPos)
+    {
+        if (args.length > 2 && args.length <= 5)
+        {
+            return getTabCompletionCoordinate(args, 2, targetPos);
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public int getRequiredPermissionLevel()
+    {
+        return 2;
+    }
+
+    @Override
+    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException
+    {
+        // args: <entity> <dim> [<x> <y> <z>]
+        if (args.length != 2 && args.length != 5)
+        {
+            throw new WrongUsageException("commands.forge.setdim.usage");
+        }
+        Entity entity = getEntity(server, sender, args[0]);
+        if (!checkEntity(entity))
+        {
+            throw new CommandException("commands.forge.setdim.invalid.entity", entity.getName());
+        }
+        int dimension = parseInt(args[1]);
+        if (!DimensionManager.isDimensionRegistered(dimension))
+        {
+            throw new CommandException("commands.forge.setdim.invalid.dim", dimension);
+        }
+        if (dimension == entity.dimension)
+        {
+            throw new CommandException("commands.forge.setdim.invalid.nochange", entity.getName(), dimension);
+        }
+        BlockPos pos = args.length == 5 ? parseBlockPos(sender, args, 2, false) : sender.getPosition();
+        entity.changeDimension(dimension, new CommandTeleporter(pos));
+    }
+
+    private static boolean checkEntity(Entity entity)
+    {
+        // use vanilla portal logic, try to avoid doing anything too silly
+        return !entity.isRiding() && !entity.isBeingRidden() && entity.isNonBoss();
+    }
+
+    private static class CommandTeleporter implements ITeleporter
+    {
+        private final BlockPos targetPos;
+
+        private CommandTeleporter(BlockPos targetPos)
+        {
+            this.targetPos = targetPos;
+        }
+
+        @Override
+        public void placeEntity(World world, Entity entity, float yaw)
+        {
+            entity.setPosition(targetPos.getX() + 0.5, targetPos.getY(), targetPos.getZ() + 0.5);
+            // clear out space for entity
+            AxisAlignedBB aabb = entity.getEntityBoundingBox();
+            BlockPos min = new BlockPos(MathHelper.floor(aabb.minX), MathHelper.floor(aabb.minY), MathHelper.floor(aabb.minZ));
+            BlockPos max = new BlockPos(MathHelper.ceil(aabb.maxX),  MathHelper.ceil(aabb.maxY),  MathHelper.ceil(aabb.maxZ));
+            for (BlockPos pos : BlockPos.getAllInBoxMutable(min, max))
+            {
+                world.setBlockToAir(pos);
```

Comment:
This can destroy important blocks, like in someone's base. What does the vanilla teleport do?

Reply:
The vanilla teleport just dumps entities at the position. I don't mind changing this, the reason for it being this way is mainly because otherwise it can be difficult to get an entity to the destination alive.

It is worth noting that the command here has a required permission level of 2, which also allows `/fill`, which can do the same. The expectation here is that as it requires operator privileges, users will be responsible/prepared to fix the consequences.

Reply:
I think by default the entity should just suffocate, if someone wants to intentionally teleport to a spot that's occupied then they can clear it with a command first. The risk of voiding a storage system or something seems a bit too scary.